File: searching/linear-search/linearSearch.go
package main

import "fmt"

func linearSearch(slc []int, target int) int {
	for index, ele := range slc {
		if ele == target {
			return index
		}
	}

	return -1
}

func main() {
	slc := []int{3, 4, 1, 6, 9, 5}
	fmt.Println(linearSearch(slc, 6))
}

File: searching/linear-search/linear_search.rs
fn linear_search(arr: &Vec<i32>, target: i32) -> i32 {
    let mut count = 0;
    for i in arr {
        if *i == target {
            return count;
        }
        count += 1;
    }
    return -1;
}
fn main() {
    let arr = vec!(1, 7, 4, 2, 9);

    println!("{}", linear_search(&arr, 4))
}